### PR TITLE
Add EmailDeliveryFailure view 

### PR DESCRIPTION
**Pull request by NikkiWines**
### Details
Adds a new "email delivery failed" view to newDot for when users are blocked because AWS / Mailgun haven't been able to deliver emails to them.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/286767
PROPOSAL: N/A

### Tests
1. Login into your account on oldDot and add the email delivery failed NVP by pasting the following in the JS console: `NVP.set('expensify_emailDeliveryFailure', {'lastSMTPErrorDatetime': moment.now().addDays(1).toString()});`
2. In an incognito window, try to log into your account in newDot 
4. Confirm you see the following view: 
     image
5. Click the `here` link and confirm it redirects to https://community.expensify.com/discussion/5651/deep-dive-best-practices-when-youre-running-into-trouble-receiving-emails-from-expensify/p1?new=1
6. Click `Back` and confirm it takes you back to login view to enter an email/sms number 
7. In your original window run `NVP.set('expensify_emailDeliveryFailure', null);` in the JS console to unblock your account

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Login into your account on oldDot and add the email delivery failed NVP by pasting the following in the JS console: `NVP.set('expensify_emailDeliveryFailure', {'lastSMTPErrorDatetime': moment.now().addDays(1).toString()});`
2. In an incognito window, try to log into your account in newDot 
3. Disable your network connection
4. Confirm you see the following view: 
Screenshot 2023-06-06 at 14 58 53

### QA Steps
1. Login into your account on oldDot and add the email delivery failed NVP by pasting the following in the JS console: `NVP.set('expensify_emailDeliveryFailure', {'lastSMTPErrorDatetime': moment.now().addDays(1).toString()});`
2. In an incognito window, try to log into your account in newDot 
4. Confirm you see the following view: 
     image
5. Click the `here` link and confirm it redirects to https://community.expensify.com/discussion/5651/deep-dive-best-practices-when-youre-running-into-trouble-receiving-emails-from-expensify/p1?new=1
6. Click `Back` and confirm it takes you back to login view to enter an email/sms number 
7. In your original window run `NVP.set('expensify_emailDeliveryFailure', null);` in the JS console to unblock your account

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

image

</details>

<details>
<summary>Mobile Web - Chrome</summary>

Screenshot_1685749319

</details>

<details>
<summary>Mobile Web - Safari</summary>

Simulator Screen Shot - iPhone 14 - 2023-06-02 at 16 39 49

</details>

<details>
<summary>Desktop</summary>

Screenshot 2023-06-02 at 16 17 20

</details>

<details>
<summary>iOS</summary>

Simulator Screen Shot - iPhone 14 - 2023-06-02 at 16 38 59

</details>

<details>
<summary>Android</summary>

image

</details>

### CODE REVIEW

**Comment by NikkiWines** [comment]
Some changes going on with the `SignInPage` where it was converted to a functional component and then reverted so going to hold off on updates for now

**Comment by NikkiWines** [comment]
Looks like there's a [WIP PR](https://github.com/Expensify/App/pull/20708) to migrate that component again so holding off until that's merged

**Comment by NikkiWines** [comment]
https://github.com/Expensify/App/pull/20708 is in review so going to continue to hold off on making changes

File: src/pages/signin/SignInPage.js
Comment (Santhosh-Sellavel) [inline src/pages/signin/SignInPage.js:82]: Ah, too many checks 😱

File: src/pages/signin/EmailDeliveryFailurePage.js
Comment (Santhosh-Sellavel) [inline src/pages/signin/EmailDeliveryFailurePage.js:84]: Display name is wrong

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]

**Comment by Santhosh-Sellavel** [comment]
@NikkiWines did a quick test seems to work well. I get to see the failure view.

Screenshot 2023-07-05 at 12 29 08 AM


#### There are a couple of related prop errors/warnings please fix that, thanks

Screenshot 2023-07-05 at 12 29 23 AM

Screenshot 2023-07-05 at 12 29 38 AM

**Comment by NikkiWines** [comment]
thanks for the review @Santhosh-Sellavel, updated!

**Comment by Santhosh-Sellavel** [comment]
@NikkiWines Seems the warning still occurs please check the video below

https://github.com/Expensify/App/assets/85645967/a7d20f56-06cc-47cb-9198-8a215b10203e

File: src/pages/signin/SignInPage.js
Comment (NikkiWines) [inline src/pages/signin/SignInPage.js:130]: We were somehow returning true for `shouldShowResendValidationForm` (which I think should probably be removed) but never had any logic for setting the `welcomeHeader` or `welcomeText` for that view. Now we do 😅

**Comment by NikkiWines** [comment]
Ah, sorry about that @Santhosh-Sellavel, should be good now

File: src/pages/signin/EmailDeliveryFailurePage.js
Comment (Santhosh-Sellavel) [inline src/pages/signin/EmailDeliveryFailurePage.js:87]: Can we use `useLocalizeHook` here instead of `withLocalize` HOC

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]

**Comment by NikkiWines** [comment]
Updated once more 🙇

File: src/pages/signin/EmailDeliveryFailurePage.js
Comment (NikkiWines) [reply src/pages/signin/EmailDeliveryFailurePage.js]: Yep, good catch!

**Comment by Santhosh-Sellavel** [comment]
@NikkiWines Keyboard Doesn't disappear automatically on iOS alone.

https://github.com/Expensify/App/assets/85645967/d7de6228-87b1-496f-9c7f-e9cf5b261bcd

**Comment by Santhosh-Sellavel** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [ ] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web & Desktop </summary>

https://github.com/Expensify/App/assets/85645967/66d1e0a4-dc06-4959-bfc6-c2d40d63703a

</details>

<details>
<summary>Mobile Web - Chrome & Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/85645967/89585d8c-8c9a-4bbb-b604-6b659f0fe03b

</details>

<details>
<summary>iOS & Android</summary>

https://github.com/Expensify/App/assets/85645967/777e720e-bf74-40ff-a433-82437729b084

</details>

**Review by Santhosh-Sellavel — COMMENTED** [review]
Other than [this](https://github.com/Expensify/App/pull/19508#issuecomment-1630984926) LGTM, tests well!

**Comment by NikkiWines** [comment]
@Santhosh-Sellavel added logic to dismiss the keyboard

File: src/pages/signin/EmailDeliveryFailurePage.js
Comment (Santhosh-Sellavel) [inline src/pages/signin/EmailDeliveryFailurePage.js:40]: Let's add a comment why this was added, as it look unnecessary here

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]

File: src/pages/signin/EmailDeliveryFailurePage.js
Comment (NikkiWines) [reply src/pages/signin/EmailDeliveryFailurePage.js:41]: Done!

**Review by Santhosh-Sellavel — APPROVED** [review]
LGTM tests well!

**Review by PauloGasparSv — APPROVED** [review]
LGTM

<details>
<summary>Web</summary>

image
image

</details>

<details>
<summary>Mobile Web - Chrome</summary>

Skipped, having trouble with the android emulator. Stuck on the loading screen on chrome mobile

</details>

<details>
<summary>Mobile Web - Safari</summary>

image

</details>

<details>
<summary>Desktop</summary>
image

image

</details>

<details>
<summary>iOS</summary>

image

</details>

<details>
<summary>Android</summary>

image

</details>

**Comment by mountiny** [comment]
There is a regression in the back end so going to fix that

**Comment by NikkiWines** [comment]
@Santhosh-Sellavel issue created! Sorry about that - https://github.com/Expensify/App/issues/22910

**Pull request merged**
